src/app/clusters/level-control/level-control.cpp

```diff
@@ -960,18 +960,56 @@ static void moveHandler(app::CommandHandler * commandObj, const app::ConcreteCom
         goto send_default_response;
     }
 
+    // Always validate the rate parameter received. A rate of 0 is invalid.
     if (!rate.IsNull() && (rate.Value() == 0))
     {
-        // Move at a rate of zero is no move at all. Immediately succeed without touching anything.
-        ChipLogProgress(Zcl, "Immediate success due to move rate of 0 (would move at no rate).");
-        status = Status::Success;
+        // Providing rate of 0 is not allowed.
+        status = Status::InvalidCommand;
         goto send_default_response;
     }
 
+    uint8_t eventDuration; // use this local var so state->eventDurationMs is only set once the command is validated.
+#ifndef IGNORE_LEVEL_CONTROL_CLUSTER_TRANSITION
+    // If the Rate field is null, the device should move at the default move rate, if available,
```

Comment:
I think this should be outisde the ifndef since it describes the #if-else behavior, no?

Reply:
no, this only describes the condition we use the default rate if the rate is null. Outside the `#ifndef IGNORE_LEVEL_CONTROL_CLUSTER_TRANSITION`, it ignores transition and runs at max speed regardless of rate or default rate setting